app/src/main/java/com/example/lhi06/residue/ItunesService.java

```diff
@@ -16,22 +16,33 @@
     private final ItunesApi service;
 
      ItunesService() {
-        Retrofit retrofit = new Retrofit.Builder()
+        final Retrofit retrofit = new Retrofit.Builder()
                 .baseUrl(BASE_API_URL)
                 .addConverterFactory(GsonConverterFactory.create())
                 .build();
 
         service = retrofit.create(ItunesApi.class);
     }
 
-    void getArtistsAlbums(@NonNull final String artistName, Callback<AlbumSearchResponse> callback) {
+    void getArtistsAlbums(@NonNull final String artistName, final Callback<AlbumSearchResponse> callback) {
         service.getArtistsAlbums(artistName).enqueue(callback);
     }
 
+    void getAlbumTracks(@NonNull final String collectionId, final Callback<AlbumTrackListResponse> callback) {
+        service.getAlbumTracks(collectionId).enqueue(callback);
+    }
+
     interface ItunesApi {
 
         @GET("/search?entity=album&attribute=artistTerm")
         Call<AlbumSearchResponse> getArtistsAlbums(@Query("term") String artistName);
+
+        @GET("/lookup?entity=song")
+        Call<AlbumTrackListResponse> getAlbumTracks(@Query("id") String collectionId);
     }
 
 }
+//
+//https://itunes.apple.com/lookup?id=211192863
+//
+//        https://itunes.apple.com/search?term=guns and roses&entity=album&attribute=artistTerm
```

Comment:
If you want to keep this comment for reference purposes then add a comment to describe what it is e.g. "Example URL" or something